### ISSUE TITLE
Feature Request - Notify unread chat messages

### ISSUE THREAD

**Issue by italoaguiar** [Azure/Communication#400]
**Description**
Provide a way to notify users for unread chat messages (like MS Teams)

**Suggestion**
*   New Event Grid event (for general devices)
*   Push Notification (for mobile devices)

**Note**
This is not a bug but the UserVoice form is unavailable (404)

**Comment by ashwinder** [comment]
Hi @italoaguiar, Can you help answer few questions so we can understand whats the use case you are enabling. 
1. What platform are you using for server & client ? (eg .Net, JavaScript, etc) 
2. What's the use case - Do you want to send a push notification when the user is not active on the app or you want to update the app UI while the user is active in the app?  
3. Are you using [real time notifications](https://docs.microsoft.com/en-us/azure/communication-services/concepts/chat/concepts#real-time-notifications) and read receipts in your application endpoint to subscribe and display new messages?

**Comment by ashwinder** [comment]
Hi @italoaguiar, Can you help answer few questions so we can understand whats the use case you are enabling. 
1. What platform are you using for server & client ? (eg .Net, JavaScript, etc) 
2. WHats the use case - Do you want to send a push notification while the app 
3. Are you using [real time notifications](https://docs.microsoft.com/en-us/azure/communication-services/concepts/chat/concepts#real-time-notifications) and read receipts in your application endpoint to subscribe and display new messages?

**Comment by italoaguiar** [comment]
Hi @ashwinder ,

1. We use .net Core 3.1 for our back-end and react native for web and mobile devices (using ACS UI Lib)
2. We need to notify new messages when a user is not active in the chat. Notifications may be through push, email and sms.
3. Yes and No. It's complex to keep a connection opened for every chat thread to receive real time read events. We have a big number of chat threads in use. We are using a timed service to check if a message is read.

Our implementation:
![image](https://pharmarocket.blob.core.windows.net/arquivos/flow.png)

A new read event in EventGrid can help us to simplify our implementation. But the timed service still checking the database for unread messages to send the notifications. It's nice to have a complete notification event.

In other words, Microsoft Teams sends a email when the offline user receive new messages. We need this feature.

**Comment by ProbablePrime** [comment]
This has been migrated to: https://feedback.azure.com/d365community/idea/1019a176-d730-ec11-a819-000d3ae2b306

**Comment by angiurgiu** [comment]
Hello Italo Almeida Aguiar,

Thank you so much for the detailed explanation of your scenario and your architecture. It has helped us understand your requirement clearly now. Just one final question, do you also want to notify the users if they are offline, or only in the case that they haven't sent a read receipt for the message (regardless of online/offline status).

We are currently brainstorming on the best way to solve your issue.

Thank you,
Andrei

**Comment by italoaguiar** [comment]
Hi @angiurgiu,

It's nice to have a way to retrieve user status (online/offline). It's a great suggestion to insert in project backlog.

In our project, when user is active, we play a sound for every message received. This solve our issue for online users.

We use ACS to provide a communication channel between users and sellers. When the user sends message outside business hours, we need to notify a next day seller response (email, sms, push notification). Our timed service provide a nice delay before notify to ensure a minimal time for user read. 

I want to simplify the archtecture. A simple EventGrid event for read receipt can do it. But a new event triggered when any message is unread for more than 1 minute can eliminate completly our uneficient timed service.

**Comment by angiurgiu** [comment]
Thanks for answering our questions. We've taken your feature request into our backlog and will consider it in the future.
